src/main/java/net/minecraftforge/items/ItemHandlerHelper.java

```diff
@@ -197,36 +138,126 @@ public static void giveItemToPlayer(EntityPlayer player, @Nonnull ItemStack stac
         }
     }
 
-    /**
-     * This method uses the standard vanilla algorithm to calculate a comparator output for how "full" the inventory is.
-     * This method is an adaptation of Container#calcRedstoneFromInventory(IInventory).
-     * @param inv The inventory handler to test.
-     * @return A redstone value in the range [0,15] representing how "full" this inventory is.
-     */
-    public static int calcRedstoneFromInventory(@Nullable IItemHandler inv)
+    public static int getFreeSpaceForSlot(IItemHandler handler, int slot)
```

Comment:
why is this not just a method on IItemHandler

Reply:
had it in there, will move it back